Microsoft.Toolkit.Uwp.UI.Media/Geometry/Parsers/ColorParser.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Numerics;
+using System.Text.RegularExpressions;
+using Microsoft.Toolkit.Uwp.UI.Media.Geometry.Core;
+using Windows.UI;
+
+namespace Microsoft.Toolkit.Uwp.UI.Media.Geometry.Parsers
+{
+    /// <summary>
+    /// Parser for Color
+    /// </summary>
+    internal static class ColorParser
+    {
+        /// <summary>
+        /// Converts the color string in Hexadecimal or HDR color format to the
+        /// corresponding Color object.
+        /// The hexadecimal color string should be in #RRGGBB or #AARRGGBB format.
+        /// The '#' character is optional.
+        /// The HDR color string should be in R G B A format.
+        /// (R, G, B &amp; A should have value in the range between 0 and 1, inclusive)
+        /// </summary>
+        /// <param name="colorString">Color string in Hexadecimal or HDR format</param>
+        /// <returns>Color</returns>
+        internal static Color Parse(string colorString)
+        {
+            var match = RegexFactory.ColorRegex.Match(colorString);
+            if (!match.Success)
+            {
+                throw new ArgumentException("Invalid Hexadecimal string!", nameof(colorString));
+            }
```

Comment:
This could just be `Guard.IsTrue(match.Success, ...)`

Reply:
Done. 
And made similar changes to other Parser classes within the namespace.